common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/FuzzyDate.scala

```diff
@@ -25,84 +24,13 @@ case class CenturyAndDecade(century: Int, decade: Int) extends FuzzyDate
 case class FuzzyDateRange[F <: FuzzyDate, T <: FuzzyDate](from: F, to: T)
     extends TimePeriod
 
-/**
-  *  Type class for conversion of TimePeriod types to InstantRange
-  */
-trait ToInstantRange[T <: TimePeriod] {
-
-  def apply(value: T): InstantRange
-
-  def safeConvert(value: T): Option[InstantRange] =
-    Exception.catching(classOf[DateTimeException]) opt apply(value)
-}
-
-object ToInstantRange extends DateHelpers {
-
-  def convert[T <: TimePeriod](f: T => InstantRange) = new ToInstantRange[T] {
-    def apply(value: T): InstantRange = f(value)
-  }
-
-  implicit val convertCalendarDate =
-    convert[CalendarDate](value =>
-      InstantRange(localDate(value), localDate(value), value.label))
-
-  implicit val convertYear =
-    convert[Year](value =>
-      InstantRange(yearStart(value.year), yearEnd(value.year), value.label))
-
-  implicit val convertMonthAndYear =
-    convert[MonthAndYear](
-      value =>
-        InstantRange(
-          monthStart(value.month, value.year),
-          monthEnd(value.month, value.year),
-          value.label))
-
-  implicit val convertCentury =
-    convert[Century](value =>
-      convertYearRange(centuryToFuzzyDateRange(value.century)))
-
-  implicit val convertCenturyAndDecade =
-    convert[CenturyAndDecade](
-      value =>
-        convertYearRange(
-          centuryAndDecadeToFuzzyDateRange(value.century, value.decade)))
-
-  implicit val convertYearRange =
-    convert[FuzzyDateRange[Year, Year]](
-      value =>
-        InstantRange(
-          yearStart(value.from.year),
-          yearEnd(value.to.year),
-          value.label))
-
-  implicit val convertCalendarDateRange =
-    convert[FuzzyDateRange[CalendarDate, CalendarDate]](value =>
-      InstantRange(localDate(value.from), localDate(value.to), value.label))
-
-  implicit val convertMonthRangeAcrossYears =
-    convert[FuzzyDateRange[MonthAndYear, MonthAndYear]](
-      value =>
-        InstantRange(
-          monthStart(value.from.month, value.from.year),
-          monthEnd(value.to.month, value.to.year),
-          value.label))
-
-  implicit val convertMonthRangeWithinAYear =
-    convert[FuzzyDateRange[Month, MonthAndYear]](
-      value =>
-        InstantRange(
-          monthStart(value.from.month, value.to.year),
-          monthEnd(value.to.month, value.to.year),
-          value.label))
-
-  implicit val convertDayRangeWithinAMonth =
-    convert[FuzzyDateRange[Day, CalendarDate]](
-      value =>
-        InstantRange(
-          localDate(value.from.day, value.to.month, value.to.year),
-          localDate(value.to),
-          value.label))
+object FuzzyDateRange {
+  def combine[
+    FF <: FuzzyDate,
+    TT <: FuzzyDate
```

Comment:
Can we get nicer / more descriptive type parameter names here?

Reply:
Actually since I'm using an existential type for the unused params (rather than `FT`, `TF` as I think I'd originally intended) would `F` and `T` to match existing/other code suffice?

Reply:
I think descriptive type parameters make code much more readable: having a `T` is fine when it's really clear from the context (e.g. `Vec[T]`), but when it's not, and especially when there is more than one, imo descriptive is much more preferable. I still have no idea what `T` and `F` mean.

I have just looked at the code though and see that it was me who originally wrote `T` and `F`, so take the blame for that :see_no_evil:

Reply:
`T`o and `F`rom - they are used so extensively (and consistently) that I think I'll keep them as is. If you're in that area of the codebase you quickly get used to them.

Reply:
Yep I guessed that meaning, OK sounds good :+1: